diff --git a/node_modules/node-gyp/lib/find-visualstudio.js b/node_modules/node-gyp/lib/find-visualstudio.js
index 16f6e79..100e42d 100644
--- a/node_modules/node-gyp/lib/find-visualstudio.js
+++ b/node_modules/node-gyp/lib/find-visualstudio.js
@@ -138,7 +138,7 @@ VisualStudioFinder.prototype = {
     ]
 
     this.log.silly('Running', ps, psArgs)
-    var child = execFile(ps, psArgs, { encoding: 'utf8' },
+    var child = execFile(ps, psArgs, { encoding: 'utf8', maxBuffer: 32 * 1024 * 1024 },
       (err, stdout, stderr) => {
         this.parseData(err, stdout, stderr, cb)
       })
@@ -262,6 +262,10 @@ VisualStudioFinder.prototype = {
       ret.versionYear = 2022
       return ret
     }
+    if (ret.versionMajor >= 18) {
+      ret.versionYear = 2022
+      return ret
+    }
     this.log.silly('- unsupported version:', ret.versionMajor)
     return {}
   },
